crates/bevy_physics/src/d3/ray.rs

```diff
@@ -0,0 +1,59 @@
+use bevy_render::camera::Camera;
+use bevy_transform::components::GlobalTransform;
+use bevy_window::Window;
+use glam::{Vec2, Vec3};
+
+pub struct Ray {
+    origin: Vec3,
+    direction: Vec3,
+}
+
+impl Ray {
+    pub fn new(origin: Vec3, direction: Vec3) -> Self {
+        direction.normalize();
+        Self { origin, direction }
+    }
+
+    pub fn from_mouse_position(
+        mouse_position: &Vec2,
```

Comment:
Window now has `.cursor_position()` method.

Reply:
Nice. I'll add it.